examples/daisyworld.jl

```diff
@@ -48,249 +60,269 @@ mutable struct Daisy <: AbstractAgent
+    agent.age += 1
+    agent.age >= model.max_age && kill_agent!(agent, model)
+end
+
+# We also need to define a version for the `Land` instances
+# (the dynamics of the `Land` are resolved at model level)
+agent_step!(agent::Land, model::DaisyWorld) = nothing
+
+# The model step function and agent step functions for Agents.jl to advance
 # Daisyworld's dynamics. Since we have constructed a number of helper functions,
 # these methods are quite straightforward.
 
-function solar_activity!(model::ABM{Daisy})
+function model_step!(model)
+    for n in nodes(model)
+        update_surface_temperature!(n, model)
+        diffuse_temperature!(n, model)
+        propagate!(n, model)
+    end
+    model.tick += 1
+    solar_activity!(model)
+end
+
+# Notice that `solar_activity!` changes the incoming solar radiation over time,
+# if the given "scenario" (a model parameter) is `:ramp`.
+# The parameter `tick` of the model keeps track of time.
+
+function solar_activity!(model::DaisyWorld)
     if model.scenario == :ramp
         if model.tick > 200 && model.tick <= 400
-            model.solar_luminosity += 0.005
+            model.solar_luminosity += model.solar_change
         end
         if model.tick > 500 && model.tick <= 750
-            model.solar_luminosity -= 0.0025
+            model.solar_luminosity -= model.solar_change/2
         end
+    elseif model.scenario == :change
+        model.solar_luminosity += model.solar_change
     end
 end
 
-function model_step!(model::ABM{Daisy})
-    for n in nodes(model)
-        suface_temperature!(n, model)
-        diffuse_temperature!(n, model)
-        propagate!(n, model)
+# ## Initialising Daisyworld
+
+# Here, we construct a function to initialize a Daisyworld. We use [`fill_space!`](@ref)
+# to fill the space with `Land` instances. Then, we need to know how many
+# daisies of each type to seed the planet with and what their albedo's are.
+# We also want a value for surface albedo, as well as solar intensity
+# (and we also choose between constant or time-dependent intensity with `scenario`).
+
+import StatsBase
+import DrWatson: @dict
+
+function daisyworld(;
+        griddims = (30, 30),
+        max_age = 25,
+        init_white = 0.2, # % cover of the world surface of white breed
+        init_black = 0.2, # % cover of the world surface of black breed
+        albedo_white = 0.75,
+        albedo_black = 0.25,
+        surface_albedo = 0.4,
+        solar_change = 0.005,
+        solar_luminosity = 1.0, # initial luminosity
+        scenario = :default,
+    )
+
+    space = GridSpace(griddims, moore = true, periodic = true)
+    properties = @dict max_age surface_albedo solar_luminosity solar_change scenario
+    properties[:tick] = 0
+    ## create a scheduler that only schedules Daisies
+    daisysched(model) = [a.id for a in allagents(model) if a isa Daisy]
+    model = ABM(Union{Daisy, Land}, space;
+        scheduler = daisysched, properties = properties, warn = false
```

Comment:
With all the mixed models we're building, I wonder if we should go back and take a look at this warning flag. Another issue for another time though.

Reply:
I think the flag is fine. Not a big deal to add `warn = false`.